### ISSUE TITLE
JDL import: ManyToMany relationship generates wrong relationship table name

### ISSUE THREAD

**Issue by postpersonality** [jhipster/jhipster-core#419]
##### **Overview of the issue**

I use JDL file
```
entity Material {
    name String
}
entity MaterialOrigin {
    name String
}
relationship ManyToMany {
    Material{origin} to MaterialOrigin{material}
}
```

This wrongly generates relationship table name as "material_origin" instead of "material_origin_material" in files:
* `.../domain/Material.java`
* `.../changelog/..._added_entity_constraints_Material.xml`
* `.../changelog/..._added_entity_Material.xml`

The issue reveals itself on `./mvnw` with the error
```
WARN 26590 --- [  restartedMain] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is java.lang.RuntimeException: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'servletEndpointRegistrar' defined in class path resource [org/springframework/boot/actuate/autoconfigure/endpoint/web/ServletEndpointManagementContextConfiguration$WebMvcServletEndpointManagementContextConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.endpoint.web.ServletEndpointRegistrar]: Factory method 'servletEndpointRegistrar' threw exception; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'auditEventsEndpoint' defined in class path resource [org/springframework/boot/actuate/autoconfigure/audit/AuditEventsEndpointAutoConfiguration.class]: Unsatisfied dependency expressed through method 'auditEventsEndpoint' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'customAuditEventRepository' defined in file [.../target/classes/com/company/app/repository/CustomAuditEventRepository.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'persistenceAuditEventRepository': Cannot create inner bean '(inner bean)#1c95bc62' of type [org.springframework.orm.jpa.SharedEntityManagerCreator] while setting bean property 'entityManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#1c95bc62': Cannot resolve reference to bean 'entityManagerFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is org.hibernate.MappingException: Foreign key (FKipo7e0wsvsnrag9hoy1hnmeb6:material_origin [origin_id])) must have same number of columns as the referenced primary key (material_origin [material_id,origin_id])
```

##### **Motivation for or Use Case**

The app fails to run without a proper relation table. The `material_origin_material` is not even described anywhere despite being referenced from  `.../changelog/..._added_entity_MaterialOrigin.xml`.

##### **Reproduce the error**

1. Initialize fresh app with `jhipster` and all the defaults (just hit enter like 10 times) 
2. Save the JDL above to the `jhipster-many-to-many-bug.jdl`
3. Bootstrap models with `jhipster import-jdl ./jhipster-many-to-many-bug.jdl --force`
4. `./mvnw`

##### **Related issues**

I didn't find any.

##### **JHipster Version(s)**

6.5.1

##### **JHipster configuration**

INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
##### **JHipster Version(s)**

```
jhipster@0.0.1-SNAPSHOT /home/nikolay.seliverstov/Projects/jh-manytomany-bug
└── generator-jhipster@6.5.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.janssen.fsra",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.5.1",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.janssen.fsra",
    "packageFolder": "com/janssen/fsra",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": []
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Material {
  name String
}
entity MaterialOrigin {
  name String
}
relationship ManyToMany {
  Material{origin} to MaterialOrigin{material}
}
</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.5" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.5+10, mixed mode)

git version 2.17.1

node: v8.15.1

npm: 6.4.1

yeoman: 3.1.1

Docker version 19.03.5, build 633a0ea838

docker-compose version 1.23.2, build 1110ad01

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>Material.json</summary>
<pre>
{
    "name": "Material",
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String"
        }
    ],
    "relationships": [
        {
            "relationshipType": "many-to-many",
            "otherEntityName": "materialOrigin",
            "otherEntityRelationshipName": "material",
            "relationshipName": "origin",
            "otherEntityField": "id",
            "ownerSide": true
        }
    ],
    "changelogDate": "20191220115447",
    "entityTableName": "material",
    "dto": "no",
    "pagination": "no",
    "service": "no",
    "jpaMetamodelFiltering": false,
    "fluentMethods": true,
    "readOnly": false,
    "clientRootFolder": "",
    "applications": "*"
}
</pre>
</details>

<details>
<summary>MaterialOrigin.json</summary>
<pre>
{
    "name": "MaterialOrigin",
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String"
        }
    ],
    "relationships": [
        {
            "relationshipType": "many-to-many",
            "otherEntityName": "material",
            "otherEntityRelationshipName": "origin",
            "relationshipName": "material",
            "otherEntityField": "id",
            "ownerSide": false
        }
    ],
    "changelogDate": "20191220115448",
    "entityTableName": "material_origin",
    "dto": "no",
    "pagination": "no",
    "service": "no",
    "jpaMetamodelFiltering": false,
    "fluentMethods": true,
    "readOnly": false,
    "clientRootFolder": "",
    "applications": "*"
}
</pre>
</details>

##### **Browsers and Operating System**

Ubuntu 18.04
Firefox 71

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by MathieuAA** [comment]
I've tried generating the entities/relationship with the CLI, the only difference I've got between the generated JSON is in MaterialOrigin.json, is the `"otherEntityField": "id"` attribute that shouldn't be here.
Apart from that, the generation is the same...

**Comment by nikolay-gb** [comment]
Well, the issue is not about difference between cli and something else. The issue that I have my own table material_origin and the framework tries to generate the many to many relationship table with the same name. Which is obviously wrong.

**Comment by MathieuAA** [comment]
In the JDL, you have the option to specify the table name.
Like that
```jdl
entity MyEntity(my_table_name)
```

**Comment by nikolay-gb** [comment]
Currently I see two workarounds. One is to rename the material_origin table to something like MaterialOrigin(material_origin_name). The second one is to change the owner of the relationship.  But that's usually not desirable. Personally I go simply fixing the generated files

**Comment by MathieuAA** [comment]
> The issue that I have my own table material_origin and the framework tries to generate the many to many relationship table with the same name. Which is obviously wrong.

Do you mean that you had an existing table in your DB and you used JHipster on top of the same schema?

**Comment by nikolay-gb** [comment]
No, I'm not. The entity Material Origin generates table with the name material_origin and the many to many relationship generates a table with the exact same name. The compilation is ok but these explode in runtime.

**Comment by MathieuAA** [comment]
Moving it back to the generator, it's a deeper issue than just JDL generation.
